Sources/SwiftParser/Nominals.swift

```diff
@@ -352,18 +352,10 @@ extension Parser {
         )
       } while keepGoing != nil && loopProgress.evaluate(currentToken)
     }
-    let unexpectedBeforeRangle: RawUnexpectedNodesSyntax?
-    let rangle: RawTokenSyntax
-    if self.currentToken.starts(with: ">") {
-      unexpectedBeforeRangle = nil
-      rangle = self.consumePrefix(">", as: .rightAngle)
-    } else {
-      (unexpectedBeforeRangle, rangle) = self.expect(.rightAngle)
-    }
+    let rangle = self.expectWithoutRecovery(prefix: ">", as: .rightAngle)
```

Comment:
Did you remove the recovery because you found it didn't actually work as expected?

I was expecting us to parse `protocol Test<Foo Bar> {}` mostly properly, ie. with just `Bar` as unexpected. But that doesn't happen as the `>` in recovery is parsed as a postfix operator (not a right angle) and thus what actually happens is that we skip up to the `{`. Did you decide it wasn't worth handling that case?

Reply:
Yes, I couldn’t find any examples where the recovery kicked in. 

We certainly could have `expect(prefix:as:)` but it’s implementation is non-trivial because non of the current recovery infrastructure supports recovery a to somewhere inside a token. I filed https://github.com/apple/swift-syntax/issues/1923 and maybe we can implement it in a follow-up PR.

Reply:
That’s happening everywhere we currently expect a prefix. It would be nice to fix at some point.